http/response/response.go

```diff
@@ -5,3 +5,24 @@ type Info interface {
 	GetDescription() string
 	GetReturnCode() string
 }
+
+type CustomResponseType struct {
+	Model       any
+	ReturnCode  string
+	Description string
+}
+
+func (c CustomResponseType) GetDescription() string {
+	return c.Description
+}
+func (c CustomResponseType) GetReturnCode() string {
+	return c.ReturnCode
+}
+
+func New(model any, returnCode string, description string) Info {
```

Comment:
Hmm idk, I kept it in case some users may prefer it. If you think its not necesessary anymore, we can remove it

Reply:
Hmm do you foresee a use-case where keeping this interface may be useful? I only thought it was useful before your `CustomResponseType` provided the same behaviors in another reusable way. Now that we have this new object doing the same work i'm concerned that this may be a case of interface pollution unless there's a good reason for it being here.

Reply:
ok so after working on the examples I think I'm about to flip-flop on what I said previously said 😅 . This may just be a preference thing but i'm not sure I would want to instantiate a bunch of endpoints using the method below. It seems more verbose compared to before. 
```
endpoint.WithSuccessfulReturns([]response.Info{response.New([]models.Product{}, "200", "Product List")})
```

```
endpoint.WithSuccessfulReturns([]response.Info{models.UnsuccessfulResponse{}}),
```

So I think your idea of just keeping both is fine. In case someone prefers to use the `Response.New` function.